pkg/build/builder/sti.go

```diff
@@ -384,3 +398,15 @@ func scriptProxyConfig(build *api.Build) (*s2iapi.ProxyConfig, error) {
 	}
 	return config, nil
 }
+
+// copyArtifactSourceList copies the artifacts set in the build config to the
+// VolumeList struct in the s2iapi.Config
+func copyArtifactSourceList(artifactsMapping []api.ImageSourcePath) (volumeList []s2iapi.VolumeSpec) {
```

Comment:
I suggest to change the type of return value from `[]s2iapi.VolumeSpec` to `s2iapi.VolumeList{}`. This tiny change will respect our abstraction.

Inspired by this code: https://github.com/openshift/origin/blob/2a2a8b9dfdc437c5b210fd7b748294ff4afe87ff/pkg/build/builder/sti.go#L148-L156